plugins/studymaster-web/Readme.md

```diff
@@ -0,0 +1,107 @@
+  * `go_var` (default `go`) Name of the Bool variable set when VSM receives a `Go` message.
+  * `request_result_var` (default `request_result`) Name of the String variable set to `SUBMIT` or `CANCEL` upon user selection in the requested form.
+* Run the project
+* Open a browser at `localhost:8080`
+* To show a request form, in the VSM project execute an action like this:
+    * `web: [REQUEST var='name;surname;weight;gender;graduated' values='EnterName;EnterSurname;in kg;m,f,d;yes' type='text;text;text;radio;checkbox']`
+      * `var` is the semi-column separated list of variables to set. Each variable should be defined in the project.
+      * `type` is the semi-column separated list of types for each variable. The number of entries must equal the number of variables. Each type can be one among `text`, `radio`, or `checkbox`.
+        * `text`: This will be one input field which will have as a placeholder the text you type in values.
+        * `radio`: This will be one or multiple radio button, where only one can be selected for one variable. Write the label for the button in values.
+        * `checkbox`: This will be one checkbox, which can be checked or unchecked. Write the label for the checkbox in values.
+      * `values` is the semi-column separated list of initial text value, or the list of available values for checkboxes and radio buttons.
+* When the action is executed, the web page will update with a request form.
+  * User can click `submit` or `cancel`.
+* If `submit` is selected, the app will send a set of messages (one per variable) with the value of the variable.
+* In addition, the resulting user action will be stored in a variable `request_result`.
+  * Value will be either `SUBMIT` or `CANCEL`.
+
+Here are screenshots from an [Example Project](ExampleProject).
+
+Main project (top level)
+![Example project](ExampleProject/screenshot-main.png)
+
+The content of the "waiting" superstate
+![Example project](ExampleProject/screenshot-substate.png)
+
+The plugin configuration
+![Example project](ExampleProject/screenshot-config.png)
+
+The WebApp after the request has been issued
+![Example project](ExampleProject/screenshot-webgui.png)
+
+## How to build the App (react-studymaster in resources)
+
+The web app is based on React and was initialized with [Create React App](https://github.com/facebook/create-react-app).
+
+The JavaScript library React is being used here, you can find more information at <https://reactjs.org/>.
+
+To develop the web app you require [NodeJS](https://nodejs.org/). Tested with version 15.14 
+(On MacOS, can be installed with `brew install node@15.14.0`).
+You also need react-scripts top buil the app:
+
+    brew install node@15.14.0  # or download from the website
+    npm install react-scripts
+
+To build the App you need to:
+
+* `cd` into folder `main\resources\react-studymaster`
+* build a new React App to get the "node_modules" folder by typing into a console: `npx create-react-app my-app`.
```

Comment:
This only incidentally works.
Dependencies are installed via `npm install`, if new dependencies are added, only copying over the dependencies of a base create-react-app project will not work. 
(it is also more work)

Reply:
So, looks like this is a bug in the documentation, not in the code.
@chiragbhuvaneshwara , can you help me with this?

Reply:
I don't think the documentation is wrong. Just the wording is a bit confusing. People who know React will just skip the step of create react app because it is mentioned in the next line that it is used only the first time or for resetting the app. However, this information is not really required and I can remove it.

Reply:
It is definitely wrong. Every node package has a dependencies list. To install the dependencies you run npm install. You don't create a new package and copy over its dependencies. This only works in this case because the dependencies have not changed. However this can not be assumed

Reply:
I understand what you're saying and you're right, of course. However, the subsequent line in the original readme did mention that create react app was to be executed only once i.e. when the React project was first created. It's a bit confusing to word it that way. So I removed that and updated the readme.  

I suppose the original author's intention was to convey that the React frontend was bootstrapped with Create React App and not to instruct the users to run it each time. At least that's how I see it. But anyways, shouldn't be a problem now.